consensus/consensus.go

```diff
@@ -166,14 +166,59 @@ func (cs *consensusBase) Propose(cert hotstuff.SyncInfo) {
 }
 
 func (cs *consensusBase) OnPropose(proposal hotstuff.ProposeMsg) { //nolint:gocyclo
-	// TODO: extract parts of this method into helper functions maybe?
 	cs.logger.Debugf("OnPropose: %v", proposal.Block)
 
+	if !cs.check(proposal) {
+		return
+	}
+
+	block := proposal.Block
+
+	if qcBlock, ok := cs.blockChain.Get(block.QuorumCert().BlockHash()); ok {
+		cs.acceptor.Proposed(qcBlock.Command())
+	} else {
+		cs.logger.Info("OnPropose: Failed to fetch qcBlock")
```

Comment:
I believe this case can never happen since the existence of the qcBlock was already checked when verifying the QC.
Should we remove the whole if?